src/mgr/BaseMgrModule.cc

```diff
@@ -393,11 +394,13 @@ ceph_option_get(BaseMgrModule *self, PyObject *args)
     return nullptr;
   }
 
-  std::string value;
-  int r = g_conf().get_val(string(what), &value);
-  if (r >= 0) {
+  const Option *opt = g_conf().find_option(string(what));
```

Comment:
now that we already have `opt`, why not just query it for its value?
```c++
if (auto opt = g_conf().find_option(what); opt != nullptr) {
  auto value = opt->to_str();
  // ...
  return get_python_typed_option_value(opt->type, value);
} else {
  // ...
  Py_RETURN_NONE;
}
```

Reply:
opt->value and daemon_value are defaults; the actual values are stored in a separate map in mdconfig_t.